payments/providers/talpa_ecom.py

```diff
@@ -0,0 +1,445 @@
+            else:
+                raise ValidationError(_("A product order should have a specific place"))
+
+        meta_fields = [
+            {
+                "key": "placeLocation",
+                "value": area_address,
+                "label": place_name,
+                "visibleInCheckout": True,
+                "ordinal": "1",
+            },
+        ]
+
+        if width:
+            meta_fields.append(
+                {
+                    "key": "placeWidth",
+                    "value": str(width),
+                    "visibleInCheckout": True,
+                    "ordinal": "2",
+                }
+            )
+        if length:
+            meta_fields.append(
+                {
+                    "key": "placeLength",
+                    "value": str(length),
+                    "visibleInCheckout": True,
+                    "ordinal": "3",
+                }
+            )
+        if mooring_type:
+            meta_fields.append(
+                {
+                    "key": "placeMooring",
+                    "value": mooring_type,
+                    "visibleInCheckout": True,
+                    "ordinal": "4",
+                }
+            )
+
+        self.payload_add_to_items(payload, order, product_name, meta_fields)
+
+    def payload_add_customer(self, payload: dict, order: Order):
+        if (
+            hasattr(order, "lease")
+            and order.lease is not None
+            and order.lease.application
+        ):
+            application = order.lease.application
+            customer = {
+                "firstName": application.first_name.capitalize(),
+                "lastName": application.last_name.capitalize(),
+                "email": application.email.strip(),
+                "phone": application.phone_number.strip(),
+            }
+        else:
+            customer = {
+                "firstName": order.customer_first_name.capitalize(),
+                "lastName": order.customer_last_name.capitalize(),
+                "email": order.customer_email.strip(),
+                "phone": order.customer_phone.strip(),
+            }
+
+        payload["customer"] = customer
+
+    def handle_settle_order(self, order: Order, event_type: str) -> HttpResponse:
+        status: Optional[OrderStatus]
+        message: str
+        payment_status: str
+
+        if event_type == TALPA_ECOM_WEBHOOK_EVENT_PAYMENT_PAID:
+            status = OrderStatus.PAID
+            message = "Payment succeeded in Talpa eCommerce"
+            payment_status = TalpaEComPaymentDetails.PAYMENT_PAID
+        elif event_type == TALPA_ECOM_WEBHOOK_EVENT_ORDER_CANCELLED:
+            status = OrderStatus.REJECTED
+            message = "Payment rejected in Talpa eCommerce"
+            payment_status = TalpaEComPaymentDetails.PAYMENT_CANCELLED
+        else:
+            raise UnknownWebhookEventError(_("Unknown webhook event: ") + event_type)
```

Comment:
Would this result in http 500 error being returned to Talpa? Are there some webhooks that we should respond with 200 ok, and otherwise ignore them?

Reply:
It would return a 400. I don't know for sure what response they're expecting, but a bad request sounds like the right code for this situation.

```python
try:
    return self.handle_settle_order(order, event_type)
except UnknownWebhookEventError:
    return HttpResponseBadRequest()
```